### ISSUE TITLE
System.Type.GUID is empty on Android

### ISSUE THREAD

**Issue by M0n7y5** [dotnet/runtime#99459]
### Description

On Windows 
![F1k99tEpDe](https://github.com/dotnet/runtime/assets/17201053/87bdcfcc-abc3-482d-a576-3a95c703cde6)

On Android (Android 14 API 34)
![image](https://github.com/dotnet/runtime/assets/17201053/76d49e7e-8c63-4eae-ae4c-97d86a295fd0)

I assume this is bug since i was not able to obtain any relevant information about this kind of behavior from the official docs:
https://learn.microsoft.com/en-us/dotnet/api/system.type.guid?view=net-8.0#system-type-guid

### Reproduction Steps

Just do run this code on windows and android
```csharp
var id = typeof(YourCustomType).GUID;
```

### Expected behavior

I excpect non empty GUID values

### Actual behavior

GUID values are empty on android, possibly on other platforms too

### Regression?

no idea

### Known Workarounds

nope

### Configuration

.Net 8 MAUI Blazor Hybrid
Windows 11 22H2 22621.3155
x64

**Comment by M0n7y5** [comment]
Same for:
Android 10 API 29
Android 9 API 28

**Comment by MichalPetryka** [comment]
https://github.com/dotnet/runtime/blob/40ac297a1101ac293a1bae01f2e2e1556f7ec017/src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs#L2197-L2199

Mono seems to return an empty GUID for types without `GuidAttribute` instead of generating a random one at runtime.

**Comment by Perksey** [comment]
Is the behaviour of this actually defined anywhere where the attribute is not present?

**Comment by M0n7y5** [comment]
Would be nice to mention this behavior in docs tho. Anyways thx for clarification @MichalPetryka .

**Comment by jkotas** [comment]
> I expect non empty GUID values

Do you just expect non-empty GUID values? Or do you expect the GUIDs to be exactly the same between Windows and non-Windows? 

> Would be nice to mention this behavior in docs tho.

This behavior is more like a bug. We typically do not document bugs.

**Comment by M0n7y5** [comment]
> Do you just expect non-empty GUID values? Or do you expect the GUIDs to be exactly the same between Windows and non-Windows?

I expect exactly what i wrote ... *I expect non empty GUID values*. That means i'll be happy with any values ... Well unique value per type to be specific.
 
> This behavior is more like a bug. We typically do not document bugs.

Is it a bug tho? It looks like a mono feature/behavior. Basically strips down guids of not marked types by the GuidAttribute in order to reduce size. (i assume)

**Comment by am11** [comment]
> It looks like a mono feature/behavior

Yes, this answer (from 2011) agrees with you https://stackoverflow.com/a/8666656.

(If possible, would be good to make Type.GUID output consistent across runtimes rather than creating a third kind of output)

**Comment by jkotas** [comment]
`Type.GUID` has been designed for built-in COM interop. Any use of this property outside built-in COM interop is questionable.

It is not unusual to see people misusing Type.Guid as unique type identifier. (It is not actually unique type identifier. Nothing prevents two different .NET types to have the same COM GUID and there are valid scenarios where this can happen.) 

In Native AOT, Type.Guid generates a random guid for types without explicitly specified one to make this sort of misuse "work".

> If possible, would be good to make Type.GUID output consistent across runtimes rather than creating a third kind of output)

It would require porting somewhat complicated algorithm used by built-in COM to generate the Guid. This algorithm creates the Guid by computing MD5 hash over the type name and all methods visible to built-in COM interop. This algorithm is not compatible with trimming, so it makes the consistency argument futile for runtime variants that are typically used with trimming.

**Comment by M0n7y5** [comment]
Yeah it would be nice to have consistent behavior across platforms.
